scheduler/src/cook/kubernetes/api.clj

```diff
@@ -1334,11 +1334,25 @@
           ; * A job may have additional containers with the name aux-*
           job-status (first (filter (fn [c] (= cook-container-name-for-job (.getName c)))
                                     container-statuses))
-          pod-preempted-timestamp (some-> pod .getMetadata .getLabels (get (or (-> (config/kubernetes) :node-preempted-label) "node-preempted")))
+          {:keys [node-preempted-label pod-deletion-timeout-seconds]}
+          (config/kubernetes)
+          pod-metadata (some-> pod .getMetadata)
+          pod-preempted-timestamp
+          (some-> pod-metadata
+                  .getLabels
+                  (get (or node-preempted-label "node-preempted")))
+          pod-deletion-timestamp (some-> pod-metadata .getDeletionTimestamp)
           synthesized-pod-state
-          (if (some-> pod .getMetadata .getDeletionTimestamp)
-            {:state :pod/deleting
-             :reason "Pod was explicitly deleted"}
+          (if pod-deletion-timestamp
```

Comment:
I feel this code is better placed in the (only) place that immediately uses it. I.e., in the state machine processing for deleting pods. Putting in a variable that's used a long distance away is not as clear as colocating creation and use.

To be more specific, move this code into a new function, something like  (should-hard-delete-pod [config pod] ...) or maybe 'get-pod-deletion-age'. that's a nice pure function of its arguments.

Have the code around line 591 in the state machine call that function to decide if hard deleting is needed or not. Give a good comment there too why we do this.

Reply:
Done